### PR TITLE
Add a one workgroup argmax benchmark

### PR DESCRIPTION
**Pull request by angelz913**
This PR is based on #47. I opened a new one because the old one got stale.

### CODE REVIEW

File: benchmarks/argmax/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2020-2023 Google LLC
```
Comment (kuhar) [inline benchmarks/argmax/CMakeLists.txt:1]:
```suggestion
# Copyright 2024 Advanced Micro Devices Inc.
# Copyright 2020-2024 Google LLC
```

File: benchmarks/argmax/one_workgroup_argmax_main.cc [hunk 2]
```diff
@@ -0,0 +1,281 @@
+// Copyright 2020-2023 Google LLC
```
Comment (kuhar) [inline benchmarks/argmax/one_workgroup_argmax_main.cc:1]:
```suggestion
// Copyright 2024 Advanced Micro Devices Inc.
// Copyright 2020-2024 Google LLC
```

**Review by kuhar — COMMENTED** [review]
Looks good modulo copyright headers. For code that is primarily based on some existing benchmarks, we use dual copyright (the original author + the new one). You can see an example here: https://github.com/google/uVkCompute/blob/main/benchmarks/vmt/vmt_main.cc.

File: benchmarks/argmax/one_workgroup_argmax_subgroup.glsl [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# version 450 core
+#extension GL_EXT_control_flow_attributes : enable
+#extension GL_KHR_shader_subgroup_arithmetic : enable
+#extension GL_KHR_shader_subgroup_ballot : enable
+
+layout(local_size_x = 16, local_size_y = 1, local_size_z = 1) in;
```
Comment (kuhar) [inline benchmarks/argmax/one_workgroup_argmax_subgroup.glsl:6]: This is smaller than the native subgroup size for desktop GPU. How does this perform if we increase this to `32` or `64`?

Comment (angelz913) [reply benchmarks/argmax/one_workgroup_argmax_subgroup.glsl]: image
image

Comment (angelz913) [reply benchmarks/argmax/one_workgroup_argmax_subgroup.glsl]: I did not see any performance improvement

**Review by kuhar — APPROVED** [review]
LGTM

**Pull request merged**